src/main/java/bio/terra/tanagra/service/utils/ValidationUtils.java

```diff
@@ -0,0 +1,24 @@
+package bio.terra.tanagra.service.utils;
+
+import autovalue.shaded.com.google.common.base.Preconditions;
+import bio.terra.model.Filter;
+import bio.terra.model.RelationshipFilter;
+
+public final class ValidationUtils {
+  private ValidationUtils() {}
+
+  public static void validateApiFilter(Filter filter) {
+    // If one RelationshipFilterV2 group_by field is set, all group_by fields must be set.
+    if (filter != null && filter.getFilterType() == Filter.FilterTypeEnum.RELATIONSHIP) {
+      RelationshipFilter relationshipFilter = filter.getFilterUnion().getRelationshipFilter();
+      Preconditions.checkState(
```

Comment:
I wasn't sure what to do with the preconditions here. I think there might be the avro library used in Tanagra, but there are other precondition libraries used in TDR.

Reply:
If you look at the implementation of `Preconditions.checkState()`, it's pretty trivial and I would inline it here to avoid the added dependency. Or, we could remove this check completely since we don't really need web api argument validation for the prototype.

Reply:
So, I think actually we can just do this with the existing imports. Verifying now.